sklearn/utils/estimator_checks.py

```diff
@@ -1190,6 +1190,30 @@ def check_estimators_pickle(name, estimator_orig):
         unpickled_result = getattr(unpickled_estimator, method)(X)
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
+    # check if input with nans pickles properly
+    if name in ALLOW_NAN:
+        # set random 10 elements to be corrupted
+        X.ravel()[np.random.choice(X.size, 10, replace=False)] = np.nan
```

Comment:
Use an RNG for reproducibility,
```py
rng = np.random.RandomState(42)
mask = rng.choice(X.size ..
```

also `X.reshape(-1)` might be preferable. From the `numpy.ravel` documentation,
> When a view is desired in as many cases as possible, arr.reshape(-1) may be preferable